From 8de3c6bbfdd34ebd3a51fb284164c4b89687b3dc Mon Sep 17 00:00:00 2001
From: David Heidelberg <david@ixit.cz>
Date: Tue, 28 Apr 2026 20:27:12 +0200
Subject: [PATCH 150/212] dt-bindings: sound: Add NXP TFA98XX TFA2 amplifier
 binding

Add a YAML devicetree binding schema for the NXP/Goodix TFA98XX TFA2
family of I2C-controlled smart speaker amplifiers with on-chip DSP.

The binding covers supported chip variants (TFA9872 through
TFA9912) and documents the optional reset and IRQ GPIO properties.

Signed-off-by: David Heidelberg <david@ixit.cz>
---
 .../bindings/sound/nxp,tfa98xx.yaml           | 65 +++++++++++++++++++
 1 file changed, 65 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/sound/nxp,tfa98xx.yaml

diff --git a/Documentation/devicetree/bindings/sound/nxp,tfa98xx.yaml b/Documentation/devicetree/bindings/sound/nxp,tfa98xx.yaml
new file mode 100644
index 000000000000..3294c670919e
--- /dev/null
+++ b/Documentation/devicetree/bindings/sound/nxp,tfa98xx.yaml
@@ -0,0 +1,65 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/sound/nxp,tfa98xx.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: NXP/Goodix TFA98XX Smart Speaker Amplifier
+
+maintainers:
+  - phone-devel@vger.kernel.org
+
+description: |
+  The TFA98XX family are I2C-controlled smart speaker amplifiers featuring
+  an on-chip CoolFlux DSP and a Class-D amplifier. The driver loads a
+  vendor-supplied container firmware file (tfa98xx.cnt) to configure the DSP.
+
+allOf:
+  - $ref: dai-common.yaml#
+
+properties:
+  compatible:
+    enum:
+      - nxp,tfa9872
+      - nxp,tfa9873
+      - nxp,tfa9874
+      - nxp,tfa9875
+      - nxp,tfa9878
+      - nxp,tfa9888
+      - nxp,tfa9894
+      - nxp,tfa9912
+
+  reg:
+    maxItems: 1
+
+  reset-gpios:
+    maxItems: 1
+    description: GPIO connected to the RST pin (active high).
+
+  irq-gpios:
+    maxItems: 1
+    description: GPIO connected to the interrupt output pin (active low).
+
+  '#sound-dai-cells':
+    const: 0
+
+required:
+  - compatible
+  - reg
+
+unevaluatedProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/gpio/gpio.h>
+    i2c {
+        #address-cells = <1>;
+        #size-cells = <0>;
+
+        amplifier@34 {
+            compatible = "nxp,tfa9874";
+            reg = <0x34>;
+            reset-gpios = <&tlmm 69 GPIO_ACTIVE_HIGH>;
+            #sound-dai-cells = <0>;
+        };
+    };
-- 
2.55.0

